/*
 * This file was automatically generated by EvoSuite
 * Fri Dec 30 09:33:52 GMT 2016
 */

package account;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import account.Account;
import account.BankAccount;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.System;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, useJEE = true) 
public class Account_ESTest extends Account_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      Account account0 = new Account();
      String[] stringArray0 = new String[2];
      account0.go((String[]) null);
      account0.go(stringArray0);
      account0.go(stringArray0);
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      Account account0 = new Account();
      String[] stringArray0 = new String[2];
      account0.go((String[]) null);
      account0.go(stringArray0);
      account0.checkResult(0);
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      Account account0 = new Account();
      String[] stringArray0 = new String[6];
      account0.go(stringArray0);
      account0.checkResult(0);
      account0.go(stringArray0);
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      Account account0 = new Account();
      account0.checkResult(0);
      String[] stringArray0 = new String[6];
      account0.checkResult(0);
      account0.go(stringArray0);
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      Account account0 = new Account();
      account0.checkResult(0);
      account0.checkResult(0);
      account0.checkResult(0);
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      Account account0 = new Account();
      account0.checkResult(0);
      String[] stringArray0 = new String[3];
      account0.go(stringArray0);
      Account.Service(0, 0);
      account0.go(stringArray0);
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      Account account0 = new Account();
      account0.checkResult(0);
      String[] stringArray0 = new String[3];
      account0.go(stringArray0);
      Account.Service(0, 0);
      Account.Service(0, 0);
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      Account account0 = new Account();
      String[] stringArray0 = new String[3];
      account0.go(stringArray0);
      Account.Service(0, 0);
      Account.Service(0, 0);
      account0.go(stringArray0);
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      Account account0 = new Account();
      account0.go((String[]) null);
      account0.checkResult(0);
      // Undeclared exception!
      try { 
        account0.checkResult((-1099));
      } catch(System.SystemExitException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.evosuite.runtime.System", e);
      }
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      Account account0 = new Account();
      account0.checkResult(0);
      String[] stringArray0 = new String[6];
      account0.go(stringArray0);
      account0.checkResult(0);
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      Account account0 = new Account();
      String[] stringArray0 = new String[3];
      account0.go(stringArray0);
      Account.Service(0, 0);
      account0.go(stringArray0);
      Account.Service(0, (-2221));
  }

  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      Account account0 = new Account();
      BankAccount[] bankAccountArray0 = new BankAccount[5];
      BankAccount bankAccount0 = new BankAccount((-4264));
      bankAccountArray0[2] = bankAccount0;
      account0.accounts = bankAccountArray0;
      Account.Service(2, (-4264));
      account0.checkResult((-4264));
      // Undeclared exception!
      try { 
        account0.checkResult(3549);
      } catch(System.SystemExitException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.evosuite.runtime.System", e);
      }
  }

  @Test(timeout = 4000)
  public void test12()  throws Throwable  {
      Account account0 = new Account();
      String[] stringArray0 = new String[3];
      account0.go(stringArray0);
      Account.Service(0, 0);
      Account.Service(0, 0);
      Account.Service(0, (-2221));
  }

  @Test(timeout = 4000)
  public void test13()  throws Throwable  {
      Account.Bank_Total = 1;
      Account account0 = new Account();
      String[] stringArray0 = new String[4];
      // Undeclared exception!
      try { 
        account0.go(stringArray0);
      } catch(System.SystemExitException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.evosuite.runtime.System", e);
      }
  }

  @Test(timeout = 4000)
  public void test14()  throws Throwable  {
      Account account0 = new Account();
      account0.Bank_random = null;
      // Undeclared exception!
      try { 
        account0.go((String[]) null);
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test15()  throws Throwable  {
      Account account0 = new Account();
      account0.NUM_ACCOUNTS = (-1);
      String[] stringArray0 = new String[6];
      // Undeclared exception!
      try { 
        account0.go(stringArray0);
      } catch(NegativeArraySizeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("account.Account", e);
      }
  }

  @Test(timeout = 4000)
  public void test16()  throws Throwable  {
      Account account0 = new Account();
      account0.checkResult(0);
      // Undeclared exception!
      try { 
        Account.Service(0, 0);
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("account.Account", e);
      }
  }

  @Test(timeout = 4000)
  public void test17()  throws Throwable  {
      Account account0 = new Account();
      String[] stringArray0 = new String[1];
      account0.go(stringArray0);
      // Undeclared exception!
      try { 
        Account.Service(784, 784);
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // 784
         //
         verifyException("account.Account", e);
      }
  }
}
